CMakeLists.txt

```
@@ -0,0 +1,67 @@
+cmake_minimum_required(VERSION 3.22)
+
```

Let's place `include()`s at the top of the file
```suggestion
include(FetchContent)

```

---

@xx12345798 As we discussed in the sync-up, it will be more ideal if we checkout the other dependencies (e.g., zstd, fmt, clp, spdlog) within the CMakeLists.txt file with the FetchContent macros. On the other hand, since we wish to get the PR merged early, and the current setup is working nicely, we can do that in another cleanup PR or when we have the need to upgrade the dependencies.